gateway/rpc_storage_handler.go

```diff
@@ -1057,6 +1058,21 @@ func (r *RPCStorageHandler) ProcessKeySpaceChanges(keys []string, orgId string)
 			case NoticeDeleteAPICache.String():
 				apiIDsToDeleteCache = append(apiIDsToDeleteCache, splitKeys[0])
 				notRegularKeys[key] = true
+			case NoticeUserKeyReset.String():
+				userKeys := strings.Split(splitKeys[0], ".")
+				if len(userKeys) != 2 {
+					log.Error("Invalid user key reset format")
+					continue
+				}
+				userKeysToReset[userKeys[0]] = userKeys[1]
+				ok := r.Gw.MainNotifier.Notify(Notification{
+					Command: NoticeUserKeyReset,
+					Payload: key,
+					Gw:      r.Gw,
+				})
+				if !ok {
+					log.Error("Failed to notify other gateways about user key reset")
+				}
```

Comment:
lets move this logic to a different loop, check the conditionals above, here you might want to do something like:

`
case NoticeUserKeyReset.String():
    userKeysToReset = append(userKeysToReset, splitKeys[0])
    notRegularKeys[key] = true --> with this you just skip this key later when looping over keys
`

Then you can loop over `userKeysToReset` and execute all the other logic that is parsing data and notifying gateways. As in the fuure we need to split this whole function into something more understandable and short function